modules/vault/src/main/java/org/apache/ignite/internal/vault/VaultManager.java

```diff
@@ -171,6 +175,31 @@ public boolean bootstrapped() {
         }
     }
 
+    /**
+     * Persist node name to the vault.
+     *
+     * @param name Node name to persist.
+     * @return Future representing pending completion of the operation. Couldn't be {@code null}.
+     */
+    @NotNull public CompletableFuture<Void> putName(@NotNull String name) {
+        return put(NODE_NAME, name.getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * @return Node name, if was stored earlier.
+     * @throws IgniteInternalCheckedException If couldn't get node name from the vault.
+     */
+    public String name() throws IgniteInternalCheckedException {
+        synchronized (mux) {
+            try {
+                return new String(vaultService.get(NODE_NAME).get().value());
```

Comment:
I would wait for future's get outside of the synchronized block

Reply:
Looks like this pattern can break some invariants, which were the causes for mutex in general. As we discussed earlier - this pattern of waiting is used in all methods of VaultManager.